py/desitarget/targets.py

```diff
@@ -860,26 +909,53 @@ def calc_priority(targets, zcat, obscon, state=False):
                     if (obsconditions.mask(obscon) & pricon) != 0:
                         ii = (targets[scnd_target] & scnd_mask[name]) != 0
                         ii &= scnd_update
-                        # ADM LyA QSOs require more observations.
+
+                        # ADM scnd LyA QSOs may require more observations.
                         # ADM (zcut is defined at the top of this module).
-                        good_hiz = zgood & (zcat['Z'] >= zcut)
-                        # ADM Mid-z QSOs require more observations at low
-                        # ADM priority as requested by some secondary programs.
-                        if survey == "sv3":
+                        # ADM Main Survey decisions are made using QN/Redrock.
+                        if survey == "main":
+                            good_hiz = (zcat['Z'] >= zcut) | ((zcat['Z_QN'] >= zcut) &
+                                                              (zcat["IS_QSO_QN"] == 1))
+                            # ADM all non-LyA-QSOs need more low-priority passes
+                            # ADM in the Main Survey. The mid-z QSOs get 4 passes
+                            # ADM at this lower priority, as requested by some
+                            # ADM secondaries, which is set in calc_numobs_more.
+                            good_midz = (zcat['Z'] < zcut) & ((zcat['Z_QN'] < zcut) |
+                                                              (zcat["IS_QSO_QN"] != 1))
```

Comment:
Is `!= 1` the correct test here, or should it be `== 0`?

Reply:
This is correct. `IS_QSO_QN` can have values of -1, 0 and 1.